src/Symfony/Component/Security/Http/Firewall/ExceptionListener.php

```diff
@@ -112,6 +112,9 @@ private function handleAuthenticationException(GetResponseForExceptionEvent $eve
 
         try {
             $event->setResponse($this->startAuthentication($event->getRequest(), $exception));
+            if (method_exists($event, 'allowCustomResponseCode')) {
```

Comment:
The condition should be removed and `composer.json` (both for `symfony/security` and `symfony/security-http`) should be updated to only allow `symfony/http-kernel` 3.3+. The reason is that you are using `setStatusCode` above anyway, so the code only works for patched versions oh http-kernel.